packages/react/runtime/src/lifecycle/patch/error.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+import { backgroundSnapshotInstanceManager, snapshotManager } from '../../snapshot.js';
+
+export const ctxNotFoundType = 'Lynx.Error.CtxNotFound';
+
+const errorMsg = 'snapshotPatchApply failed: ctx not found';
+
+export interface CtxNotFoundData {
+  id: number;
+}
+
+export function sendCtxNotFoundEventToBackground(id: number): void {
+  const error = new Error(errorMsg);
+  /* v8 ignore next 3 */
+  if (!lynx.getJSContext) {
+    throw error;
+  }
+  lynx.getJSContext().dispatchEvent({
+    type: ctxNotFoundType,
+    data: {
+      id,
+    } as CtxNotFoundData,
+  });
+}
+
+export function reportCtxNotFound(data: CtxNotFoundData): void {
+  const id = data.id;
+  const instance = backgroundSnapshotInstanceManager.values.get(id);
+
+  const snapshotDef = [...snapshotManager.values.entries()]
```

Comment:
Can we use for-of over `snapshotManager.values.entries()` to:

1. Avoid creating an array
2. Bailout when snapshot is found